templates/problem/voting-controls.html

```diff
@@ -0,0 +1,218 @@
+            </div>
+        </form>
+    </div>
+
+    <div>
+        <a class="form-button" id="id_vote_button"></a>
+    </div>
+
+    {% if vote is not none %}
+        {{ vote.note|json_script('voted_note') }}
+    {% endif %}
+    <script>
+        let voted_points = null;
+        let voted_note = null;
+        {% if vote is not none %}
+            let has_voted = true;
+            voted_points = {{ vote.points }};
+            voted_note = JSON.parse(document.getElementById('voted_note').textContent);
+        {% else %}
+            let has_voted = false;
+        {% endif %}
+
+        function voteUpdate(){
+            $('#id_has_voted_prompt').show();
+            $('#id_current_vote_value').prop('innerText', voted_points);
+            $('#id_has_not_voted_prompt').hide();
+            $('#id_delete_vote_form_submit_button').show();
+            $('#id_vote_button').prop('innerText', `{{ _('Change vote') }} (`+voted_points+')');
+            $('#id_points').prop('value', '');
+            $('#id_points').prop('placeholder', voted_points);
+            $('#id_note').prop('value', voted_note);
+            $('#id_points_error_box').hide();
+            $('#id_note_error_box').hide();
+            has_voted = true;
+        }
+
+        function deleteVoteUpdate(){
+            $('#id_has_voted_prompt').hide();
+            $('#id_has_not_voted_prompt').show();
+            $('#id_delete_vote_form_submit_button').hide();
+            $('#id_vote_button').prop('innerText', `{{ _('Vote on problem points') }}`);
+            $('#id_points').prop('value', '');
+            $('#id_points').prop('placeholder', 10);
+            $('#id_note').prop('value', '');
+            $('#id_note').prop('placeholder', `{{ _("A short justification for your vote to this problem's points value.") }}`);
+            $('#id_points_error_box').hide();
+            $('#id_note_error_box').hide();
+            has_voted = false;
+        }
+
+        if (has_voted) voteUpdate();
+        else deleteVoteUpdate();
+
+        $('#id_delete_vote_form').on('submit', e => e.preventDefault());
+        $('#id_delete_vote_form_submit_button').on('click', e => {
+            e.preventDefault();
+            $.ajax({
+                url: '{{ url('delete_vote', object.code) }}',
+                type: 'POST',
+                data: $('#id_delete_vote_form').serialize(),
+                success: () => {
+                    deleteUserVote(voted_points);
+                    voted_points = null;
+                    voted_note = null;
+                    deleteVoteUpdate();
+                    // Forms are auto disabled to prevent resubmission, but we need to allow resubmission here.
+                    $('id_delete_vote_form_submit_button').removeAttr('disabled');
+                    closeLightbox('id_voting_lightbox');
+                },
+                error: data => {
+                    alert('Unable to delete vote: '+data.responseText);
```

Comment:
Spaces between operators, almost certainly.

Reply:
Alternatively you could maybe use interpolation.

Reply:
Fixed, along with a few other cases.